cibuildwheel/windows.py

```diff
@@ -115,6 +122,50 @@ def install_pypy(tmp: Path, arch: str, url: str) -> Path:
     return installation_path / "python.exe"
 
 
+def setup_setuptools_cross_compile(
+    python_configuration: PythonConfiguration,
+    python_libs_base: Path,
+    env: Dict[str, str],
+) -> None:
+    # We write to distutils_cfg for distutils-based builds because we know we
+    # currently don't have one (unless it's our own from a previous build)
+    for p in env["PATH"].split(os.pathsep):
+        distutils_cfg = Path(p) / "Lib/site-packages/setuptools/_distutils/distutils.cfg"
+        if distutils_cfg.parent.is_dir():
+            break
+    else:
+        log.warning("Did not find setuptools install to configure")
+        return
+
+    # Ensure our additional import libraries are made available, and explicitly
+    # set the platform name
+    map_plat = {"32": "win32", "64": "win-amd64", "ARM64": "win-arm64"}
+    plat_name = map_plat[python_configuration.arch]
+    # (Sorry, this file must be default/locale encoding.)
+    with distutils_cfg.open("w") as f:
+        print("[build]", file=f)
+        print("plat_name=", plat_name, file=f, sep="")
+        print("[build_ext]", file=f)
+        print("library_dirs=", python_libs_base, file=f, sep="")
+        print("plat_name=", plat_name, file=f, sep="")
+        print("[bdist_wheel]", file=f)
+        print("plat_name=", plat_name, file=f, sep="")
```

Comment:
No, the comment is the lack of `open(... encoding=)` argument, since there isn't a good backwards/forwards compatible way to do `encoding="locale"`.

Switching to dedent is fine.

```suggestion
    # (Sorry, this file must be default/locale encoding.)
    distutils_cfg.write_text(
        textwrap.dedent(
            f"""\
            [build]
            plat_name={plat_name}
            [build_ext]
            library_dirs={python_libs_base}
            plat_name={plat_name}
            [bdist_wheel]
            plat_name={plat_name}
            """
        )
    )
```